luna/featurevis/featurevis.py

```diff
@@ -118,13 +28,34 @@ def visualize_filter(image, model, layer, filter_index, iterations,
     Returns:
         tuple: loss and result image for the process
     """
+    image = tf.Variable(image)
     feature_extractor = get_feature_extractor(model, layer)
+
+    # Temporary method for random choice of
+    # transformation combination
+    choice_num = [0, 1, 2, 3]
+    augmentation = ['noise', 'blur', 'scale']
     print('Starting Feature Vis Process')
     for iteration in range(iterations):
         pctg = int(iteration / iterations * 100)
-        image = add_noise(image, noise, pctg, channels_first)
-        image = blur_image(image, blur, pctg)
-        image = rescale_image(image, scale, pctg)
+        image_aug = {'noise': trans.add_noise(image, noise, pctg),
+                     'blur': trans.blur_image(image, blur, pctg),
+                     'scale': trans.rescale_image(image, scale)}
+        num = random.choice(choice_num)
+        if num ==1:
+            ind = random.sample(augmentation, 1)
+            print(ind)
+            image = image_aug[ind[0]]
+        if num ==2:
+            ind = random.sample(augmentation, 2)
+            print(ind)
+            image = image_aug[ind[0]]
+            image = image_aug[ind[1]]
+        else:
+            image = trans.add_noise(image, noise, pctg)
+            image = trans.blur_image(image, blur, pctg)
+            image = trans.rescale_image(image, scale)
+
```

Comment:
This has some problems I think. First, you are ignoring the case where random choice is 0. But more importantly, it is not clear to me why this should be random. Maybe you can explain that to me?

Reply:
I was thinking: if we always add some amount of (i.e.) noise, this will be done at every single iteration and (since it depends on the pctg), it will increase the noisiness at every iteration. My idea was the following: randomize the combination of different transformations in order to not over-transform any feature since the model might not learn good (correct me if my understanding is wrong please). The same goes for every other transformations that use the pctg.

So I have two ideas:
- either I can find a more efficient way of writing this randomization
- or if the user wants to have a particular transformation, its amount should be random at every iteration and not depended on pctg (similar to rescale image as described in the paper which I am actually using their reported values)

The transformations have a strong effect on the end result.